examples/system_metrics/collect_system_metrics.py

```diff
@@ -0,0 +1,12 @@
+import time
+
+import mlflow
+
+if __name__ == "__main__":
+    mlflow.enable_system_metrics_logging()
+    run = mlflow.start_run()
```

Comment:
I think that `mlflow.enable_system_metrics_logging()` is redundant with the `include_system_metrics` argument in `mlflow.start_run()`, since both of these indicators just tell MLflow to log system metrics during the next fluent run that's created, right?

Instead, I was hoping that we could expose SystemMetricsMonitor() as something that users can create (https://github.com/mlflow/mlflow/pull/9557/files#diff-d6f29b7612148b947899a47b6952c2969d23a42a7bbe88cdd4605ae4c5ad853bR16) so that they can use it inside runs that *aren't* created from the fluent API.

Maybe we should have `mlflow.system_metrics.start_monitor(run_id)`?

Reply:
Yes it's a bit confusing, but not quite redundant, the control logic is:
- If the environment variable is set, we use it to determine if we turn on system metrics logging. This is useful if users want to turn on/off system metrics logging globally.
- If the environment variable is not set, we use `include_system_metrics ` (`log_system_metrics` in your proposal) to determine if we turn it on. This is useful if users don't want to turn on system metrics logging globally.

The biggest advantage of putting `include_system_metrics` in the fluent api `start_run()` over `mlflow.system_metrics.start_monitor(run_id)` is users don't need to remember to call `mlflow.system_metrics.end_monitor(run_id)` to kill the logging thread. `end_run` or using `with mlflow.start_run() as run` has been our contract for a while, so users are not required to add extra code.

Reply:
I would go with making the default parameter value for include_system_metrics=None in that case and having the impl try to read the env var. I think the mlflow.enable_system_metrics() is entirely redundant if we make that change to the default kwarg value

Reply:
`enable_system_metrics()`/`disable_system_metrics()` is a pair of API to manage environment variables, which does not affect the functionality, just provides an API for users not comfortable with configuring env variables directly.

Reply:
But, if this does the exact same thing as the fluent argument, why do we need both?

Reply:
Ohh tensorflow also has both ways exist (`run_functions_eagerly()` and `model.compile(run_eagerly=True)`).

The diff is "global control" vs "local control", like if I want to globally enable system metrics logging for my project, I will use `mlflow.enable_system_metrics()` at some `__init__.py`. But if I only want to enable system metrics for one of my workflow, I will use `mlflow.start_run(log_system_metrics=True)`.  For a single run, yes, it's unnecessary.

Reply:
I see. It's clear now: `enable_system_metrics()` enables system metrics for multiple runs. Thanks @chenmoneygithub :)